### ISSUE TITLE
Unexpected token operator '?' when using the legacy build

### ISSUE THREAD

**Issue by devinant** [mozilla/pdf.js#15961]
Attach (recommended) or Link to PDF file here: N/A

Configuration:
- Web browser and its version: N/A
- Operating system and its version: macOS Monterey v12.6 (M1)
- PDF.js version: v3.2.146 (legacy)
- Is a browser extension: No

Other (perhaps) relevant info:
- Bundler: Parcel 1.12.5
- TypeScript: 4.7.4

What is the expected behavior?:
The application should be built

What went wrong?
I am getting the following error when bundling with parcel in production mode:

```bash
/app/node_modules/pdfjs-dist/legacy/build/pdf.js: Unexpected token: operator (?)
    at Z.get (/app/node_modules/terser/dist/bundle.min.js:1:525)
    at Object.errorToJson (/app/node_modules/@parcel/utils/src/errorUtils.js:9:20)
    at Pipeline.process (/app/node_modules/parcel-bundler/src/Pipeline.js:29:26)
    at async Object.run (/app/node_modules/parcel-bundler/src/worker.js:15:12)
    at async Child.handleRequest (/app/node_modules/@parcel/workers/src/child.js:60:26)
```

The stack-trace above does not provide any more details than that. However, upon further investigation in `node_modules/pdfjs-dist/legacy/build/pdf.js` I noticed the use of more modern features, such as  the [nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) and the **let** keyword.

The [README](https://github.com/mozilla/pdfjs-dist/blob/master/README.md) in [the mozilla/pdfjs-dist repository](https://github.com/mozilla/pdfjs-dist) states:
> For usage with older browsers or environments, **_without support for modern features such as optional chaining, nullish coalescing_**, and private class fields/methods; please see the legacy/ folder.

Is this bug or is the README wrong? We are using the legacy build.

---

When the application is not bundled in production mode (no minification). Then it works as expected.

If relevant, in the application, PDF.js is imported like this:
```ts
import { getDocument } from "pdfjs-dist/legacy/build/pdf"
```

~~Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension)~~

**Comment by Snuffleupagus** [comment]
The currently supported browsers/environments can always be found in https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#faq-support and according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#browser_compatibility all of them have native support for the `??` operator, hence why that's no longer polyfilled in the `legacy`-build.

> Other (perhaps) relevant info:
>     * Bundler: Parcel 1.12.5
>     * TypeScript: 4.7.4

Most likely those tools are too old to support the necessary features, so you probably want to upgrade them to the latest releases.

**Comment by devinant** [comment]
> The currently supported browsers/environments can always be found in https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#faq-support and according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#browser_compatibility all of them have native support for the `??` operator, hence why that's no longer polyfilled in the `legacy`-build.
>
Thanks for the clarification, that sounds reasonable. But that means that the README is wrong, no? Let me know if you need a contribution to change that text.

> 
> > Other (perhaps) relevant info:
> > * Bundler: Parcel 1.12.5
> > * TypeScript: 4.7.4
> 
> Most likely those tools are too old to support the necessary features, so you probably want to upgrade them to the latest releases.

Yes. Parcel is outdated and needs to be updated to v2. Not a simple task I am afraid.

Thanks for the quick reply!

**Comment by devalberg** [comment]
I was able to make the latest `pdfjs-dist/legacy` compatible down to Safari 12.1 (iOS) and Chrome (Windows/Mac OS) v69.

In my case we are using NextJS, what we had to do was use `next-transpile-module` and add `pdfjs-dist` in our `next.config.js`. 

Additionally added this to babel config:

```
{
  "presets": [
    "next/babel",
    [
      "@babel/preset-env",
      {
        "useBuiltIns": "usage",
        "corejs": { "version": 3, "proposals": true }
      }
    ]
  ]
}
```

There are still some minor errors being logged but the pdf renders correctly in the versions mentioned.

Hopefully this helps something 🙂
